hack/gendata/cmds/options/cloud_data.go

```diff
@@ -0,0 +1,97 @@
+package options
+
+import (
+	"github.com/appscode/go/flags"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+const (
+	DefaultKubernetesVersion string = "1.8.0"
+)
+
+type CloudData struct {
+	Provider string
+	//credential file for gce
+	CredentialFile string
+	//access token for digitalocean
+	DoToken string
+	//access token for packet
+	PacketToken          string
+	GCEProjectName       string
+	AWSRegion            string
+	AWSAccessKeyID       string
+	AWSSecretAccessKey   string
+	KubernetesVersions   string
+	AzureTenantId        string
+	AzureSubscriptionId  string
+	AzureClientId        string
+	AzureClientSecret    string
+	VultrApiKey          string
+	LinodeApiKey         string
+	ScalewayToken        string
+	ScalewayOrganization string
+}
+
+func NewCloudData() *CloudData {
+	return &CloudData{
+		Provider:           "",
+		KubernetesVersions: DefaultKubernetesVersion,
+	}
+}
+
+func (c *CloudData) AddFlags(fs *pflag.FlagSet) {
```

Comment:
- [ ] Flags should be formatted like `provider.w1-w2`
- [ ] Credential constants: https://github.com/pharmer/pharmer/blob/bd87eb23b28714273621132b152dbc0ff3eb6684/credential/types.go#L14